pytensor/gradient.py

```diff
@@ -2052,6 +2052,72 @@ def hessian(cost, wrt, consider_constant=None, disconnected_inputs="raise"):
     return as_list_or_tuple(using_list, using_tuple, hessians)
 
 
+def hessian_prod(cost, wrt, p, **grad_kwargs):
+    """Return the expression of the Hessian times a vector p.
+
+    Parameters
+    ----------
+    cost: Scalar (0-dimensional) variable.
+    wrt: Vector (1-dimensional tensor) 'Variable' or list of Vectors
+    p: Vector (1-dimensional tensor) 'Variable' or list of Vectors
+        Each vector will be used for the hessp wirt to exach input variable
+    **grad_kwargs:
+        Keyword arguments passed to `grad` function.
+
+    Returns
+    -------
+    :class:` Vector or list of Vectors
+        The Hessian times p of the `cost` with respect to (elements of) `wrt`.
+
+    Examples
+    --------
+
+    .. testcode::
```

Comment:
@OriolAbril Any idea if this will actually run as a test? I duplicated the code for now as an explicit test, because I don't trust it :D

Reply:
That's too logical @Armavica

Reply:
It is executed, but there is no `testoutput` so it isnt very useful. In general the testcode directive is used followed by a testoutput to ensure executing what is inside testcode generates what is inside testoutput, otherwise the doc build fails IIRC. https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html has some examples

Reply:
There are asserts in the code

Reply:
If the goal is that this can be part of both the documentation and the docstring, then that would be the way to go. Provided that 1) `sphinx.ext.doctest` is in the list of extensions in `conf.py` and 2) pytest is configured to search for tests in docstrings and/or documentation files (no idea how this is done though).

If the goal is to have a snippet of code in the docstring that works as an example and is executed automatically when building the docs, maybe the ipython directive is better when it comes to registering and rendering generated outputs

Reply:
I don't know what the ipython directive is. I would like the test to run in the normal CI with pytest, any pointers for how to do this with sphinx testcode snippets?

Reply:
It looks like adding `--doctest-modules` to [addopts](https://github.com/pymc-devs/pytensor/blob/main/pyproject.toml#L119) will do the trick. Ref: https://docs.pytest.org/en/stable/how-to/doctest.html

Reply:
Adding to addopts was not enough, gonna try to command explicitly in our CI